geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/tables/Z3Table.scala

```diff
@@ -108,21 +108,25 @@ object Z3Table extends GeoMesaTable {
     bd.delete()
   }
 
-  private def getRowKey(ftw: FeatureToWrite, dtgIndex: Int): Array[Byte] = {
-    val geom = ftw.feature.point
-    val x = geom.getX
-    val y = geom.getY
-    val dtg = new DateTime(ftw.feature.getAttribute(dtgIndex).asInstanceOf[Date])
+  def getRowPrefix(x: Double, y: Double, time: Long): Array[Byte] = {
+    val dtg = new DateTime(time)
     val weeks = epochWeeks(dtg)
     val prefix = Shorts.toByteArray(weeks.getWeeks.toShort)
-
     val secondsInWeek = secondsInCurrentWeek(dtg, weeks)
     val z3 = SFC.index(x, y, secondsInWeek)
     val z3idx = Longs.toByteArray(z3.z)
+    Bytes.concat(prefix, z3idx)
+  }
 
+  private def getRowKey(ftw: FeatureToWrite, dtgIndex: Int): Array[Byte] = {
+    val geom = ftw.feature.point
+    val x = geom.getX
+    val y = geom.getY
+    val dtg = ftw.feature.getAttribute(dtgIndex).asInstanceOf[Date]
+    val time = if (dtg == null) System.currentTimeMillis() else dtg.getTime
```

Comment:
The SFT is guaranteed to have a dtg field, and normally all features would have a dtg, but this catches the case where a particular feature has a null date. We could just not support that case, but this way it won't throw an NPE and bomb out the whole ingest.